diff --git a/node_modules/intl/lib/core.js b/node_modules/intl/lib/core.js
index 9334cfc..4525a08 100644
--- a/node_modules/intl/lib/core.js
+++ b/node_modules/intl/lib/core.js
@@ -540,7 +540,7 @@ function createRegExpRestore() {
 
     var regExpCache = {
         lastMatch: RegExp.lastMatch || '',
-        leftContext: RegExp.leftContext,
+        leftContext: RegExp.leftContext || '',
         multiline: RegExp.multiline,
         input: RegExp.input
     },
@@ -4291,6 +4291,7 @@ defineProperty(Intl, '__applyLocaleSensitivePrototypes', {
     writable: true,
     configurable: true,
     value: function value() {
+      return null;
         defineProperty(Number.prototype, 'toLocaleString', { writable: true, configurable: true, value: ls.Number.toLocaleString });
         // Need this here for IE 8, to avoid the _DontEnum_ bug
         defineProperty(Date.prototype, 'toLocaleString', { writable: true, configurable: true, value: ls.Date.toLocaleString });
