diff --git a/node_modules/react-native/Libraries/Core/InitializeCore.js b/node_modules/react-native/Libraries/Core/InitializeCore.js
index 25377f6..4f902d9 100644
--- a/node_modules/react-native/Libraries/Core/InitializeCore.js
+++ b/node_modules/react-native/Libraries/Core/InitializeCore.js
@@ -30,7 +30,7 @@ require('./setUpGlobals');
 require('./setUpDOM');
 require('./setUpPerformance');
 require('./setUpErrorHandling');
-require('./polyfillPromise');
+// require('./polyfillPromise');
 require('./setUpRegeneratorRuntime');
 require('./setUpTimers');
 require('./setUpXHR');
